src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs

```diff
@@ -51,6 +53,7 @@ public class UseExportProviderAttribute : BeforeAfterTestAttribute
         /// the test is failed.
         /// </summary>
         private static readonly TimeSpan CleanupTimeout = TimeSpan.FromMinutes(1);
+        private static readonly FatalError.ErrorReporterHandler s_FatalErrorHandler = (e, _, _) => ExceptionDispatchInfo.Capture(e).Throw();
```

Comment:
Rather than directly throw the exception (which may or may not lead to a test failure), it would be better to collect the exceptions in a list and rethrow them as part of an `AggregateException` if necessary in `UseExportProviderAttribute.Dispose`.

Reply:
So I admit I only put this here because I didn't have another "at start" method -- this isn't tied to the MEF container at all and the lack of any unsubscribe was so that way it'd apply to other methods too that aren't using MEF. Honestly, I'm thinking this should just move to the module initializer (or a similar one) where we currently hook up the trace listener for tests to avoid swallowing things there.

I agree there's a risk a rethrow might not fail the test, I guess I'm not that concerned about that (or at least not enough to block this.)

Reply:
I've seen real-world false negatives in tests because an exception in a fire-and-forget operation was caught by the underlying task scheduler and from there never observed. If the goal of this change is to _observe_ cases where the error reporter was called, the implementation should be performed in a way that guarantees that observation. Rethrowing is not sufficient to achieve this.